profiler/profiler.go

```diff
@@ -170,8 +175,13 @@ func (p *profiler) enqueueUpload(bat batch) {
 				p.cfg.statsd.Count("datadog.profiler.go.queue_full", 1, p.cfg.tags, 1)
 				log.Warn("Evicting one profile batch from the upload queue to make room.\n")
 			default:
-				// queue is empty; contents likely got uploaded
+				// this case should be almost impossible to trigger, it would require a
+				// full p.out to completely drain within nanoseconds or extreme
+				// scheduling decisions by the runtime.
 			}
+			// p.out is guaranteed to have room for bat now since concurrent calls to
+			// enqueueUpload are verboten by the func comment.
+			p.out <- bat // 👍
```

Comment:
Should we not return after this? 
My reading of this is that the loop will continue, and enqueue `bat` a second time if we encounter this situation.

Reply:
Furthermore, do we even need to do this here? 
It looks like `enqueueUpload` is already designed to retry indefinitely. The for loop will continue until 
```
		case p.out <- bat:
			return // 👍
```

Reply:
Thanks for spotting this @knusbaum, you're completely right!

I totally missed that there is a `for` loop around all of this 🤦 . I blame the brain damage from drinking from the information firehose since joining. I'll rework the PR to remove the code change but leave the comment improvements that make sense.

Reply:
🤦 I don't know what my excuse is.  My eyes glazed over that as well.  Thanks Kyle